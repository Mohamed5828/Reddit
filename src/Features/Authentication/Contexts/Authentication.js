import { createContext, useContext } from "react";

import useLocalStorage from "./useLocalStorage";

/* Creating a context object. */
const AuthContext = createContext(null);

/**
 * The AuthProvider is a function that takes in a child component and returns a context provider that
 * has a value of an object with the following properties: user, login, logout, getToken,
 * getExpirationDate, isLoggedIn, getFullName.
 * @returns The AuthContext.Provider is being returned.
 */
export const AuthProvider = ({ children }) => {
  // Local storage for user information
  const [user, setUser] = useLocalStorage("user", "null");

  /**
   * The login function takes a user object as an argument, and then sets the user state to the
   * stringified version of the user object.
   * @param user - The user object that is returned from the login function.
   */
  const login = async (user) => {
    setUser(JSON.stringify(user));
  };

  // Handle logout
  const logout = () => {
    localStorage.clear();
    setUser(null);
  };

  // Return user's full name
//   const getFullName = () => {
//     return JSON.parse(user).fullName;
//   };

  // Return user's token
  const getToken = () => {
    return JSON.parse(user).accessToken;
  };

  // Return token's expiration date
  const getExpirationDate = () => {
    return JSON.parse(user).expiresIn;
  };

  // Is user logged in?
  const isLoggedIn = () => {
    return user !== "null";
  };
  return (
    <AuthContext.Provider
      value={{
        //user,
        login,
        logout,
        getToken,
        getExpirationDate,
        isLoggedIn,
        //getFullName,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

/**
 * It returns the value of the AuthContext object.
 * @returns The AuthContext object.
 */
export const useAuth = () => {
  return useContext(AuthContext);
};
