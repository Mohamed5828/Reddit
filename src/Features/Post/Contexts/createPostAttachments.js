import { createContext, useContext, useState } from "react";

export const createPostAttachmentsContext = createContext({});

/**
 * Context for attachments of the post to be created
 *
 * @param {React.Component} children Who can access this context
 * @returns {React.Component}
 */
export const CreatePostAttachmentsProvider = ({ children }) => {
  // Context that is used to share attachments of the post to be created
  const [createPostAttachments, setCreatePostAttachments] = useState([]);

  return (
    <createPostAttachmentsContext.Provider
      value={{
        createPostAttachments,
        setCreatePostAttachments,
      }}
    >
      {children}
    </createPostAttachmentsContext.Provider>
  );
};
/**
 * Custom hook used to provide post attachments context
 *
 * @returns {React.Context}
 */
export const useCreatePostAttachments = () => {
  return useContext(createPostAttachmentsContext);
};
