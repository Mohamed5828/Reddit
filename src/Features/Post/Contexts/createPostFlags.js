import { createContext, useContext, useState } from "react";

export const createPostFlagsContext = createContext({});

/**
 * Context for flags of the post to be created
 *
 * @param {React.Component} children Who can access this context
 * @returns {React.Component}
 */
export const CreatePostFlagsProvider = ({ children }) => {
  // Context that is used to share flags of the post to be created
  const [createPostFlags, setCreatePostFlags] = useState({
    NSFW: false,
    spoiler: false,
  });

  return (
    <createPostFlagsContext.Provider
      value={{
        createPostFlags,
        setCreatePostFlags,
      }}
    >
      {children}
    </createPostFlagsContext.Provider>
  );
};
/**
 * Custom hook used to provide post flags context
 *
 * @returns {React.Context}
 */
export const useCreatePostFlags = () => {
  return useContext(createPostFlagsContext);
};
