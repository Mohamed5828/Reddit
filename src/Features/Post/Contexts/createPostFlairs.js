import { createContext, useContext, useState } from "react";

export const createPostFlairsContext = createContext({});

/**
 * Context for flairs of the post to be created
 *
 * @param {React.Component} children Who can access this context
 * @returns {React.Component}
 */
export const CreatePostFlairsProvider = ({ children }) => {
  // Context that is used to share flairs of the post to be created
  const [createPostFlairs, setCreatePostFlairs] = useState(null);

  return (
    <createPostFlairsContext.Provider
      value={{
        createPostFlairs,
        setCreatePostFlairs,
      }}
    >
      {children}
    </createPostFlairsContext.Provider>
  );
};
/**
 * Custom hook used to provide post flairs context
 *
 * @returns {React.Context}
 */
export const useCreatePostFlairs = () => {
  return useContext(createPostFlairsContext);
};
