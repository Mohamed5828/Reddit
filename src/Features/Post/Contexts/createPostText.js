import { createContext, useContext, useState } from "react";

export const createPostTextContext = createContext({});

/**
 * Context for text of the post to be created
 *
 * @param {React.Component} children Who can access this context
 * @returns {React.Component}
 */
export const CreatePostTextProvider = ({ children }) => {
  // Context that is used to share text of the post to be created
  const [createPostText, setCreatePostText] = useState([]);

  return (
    <createPostTextContext.Provider
      value={{
        createPostText,
        setCreatePostText,
      }}
    >
      {children}
    </createPostTextContext.Provider>
  );
};
/**
 * Custom hook used to provide post text context
 *
 * @returns {React.Context}
 */
export const useCreatePostText = () => {
  return useContext(createPostTextContext);
};
