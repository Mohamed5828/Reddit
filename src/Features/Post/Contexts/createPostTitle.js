import { createContext, useContext, useState } from "react";

export const createPostTitleContext = createContext({});

/**
 * Context for title of the post to be created
 *
 * @param {React.Component} children Who can access this context
 * @returns {React.Component}
 */
export const CreatePostTitleProvider = ({ children }) => {
  // Context that is used to share title of the post to be created
  const [createPostTitle, setCreatePostTitle] = useState("");

  return (
    <createPostTitleContext.Provider
      value={{
        createPostTitle,
        setCreatePostTitle,
      }}
    >
      {children}
    </createPostTitleContext.Provider>
  );
};
/**
 * Custom hook used to provide post submit destination context
 *
 * @returns {React.Context}
 */
export const useCreatePostTitle = () => {
  return useContext(createPostTitleContext);
};
