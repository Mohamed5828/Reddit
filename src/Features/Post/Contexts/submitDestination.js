import { createContext, useContext, useState } from "react";

export const submitDestinationContext = createContext({});

/**
 * Context for post submit destination
 *
 * @param {React.Component} children Who can access this context
 * @returns {React.Component}
 */
export const SubmitDestinationProvider = ({ children }) => {
  // Context that is used to share project info between project and workspace
  const [submitDestination, setSubmitDestination] = useState(null);

  return (
    <submitDestinationContext.Provider
      value={{
        submitDestination,
        setSubmitDestination,
      }}
    >
      {children}
    </submitDestinationContext.Provider>
  );
};
/**
 * Custom hook used to provide post submit destination context
 *
 * @returns {React.Context}
 */
export const useSubmitDestination = () => {
  return useContext(submitDestinationContext);
};
