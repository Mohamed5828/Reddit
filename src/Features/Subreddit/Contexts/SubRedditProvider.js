import { createContext, useContext, useState } from "react";
const subRedditContext = createContext(null);

export const SubRedditProvider = ({children }) => {

  const [community, setCommunity] = useState(null);

  return (
    <subRedditContext.Provider value={{ community, setCommunity/*, error, loading, reload */}}>
      {children}
    </subRedditContext.Provider>
  );
};

export const useSubReddit = () => {
  return useContext(subRedditContext);
};