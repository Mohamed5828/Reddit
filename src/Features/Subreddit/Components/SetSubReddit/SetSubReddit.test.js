import Adapter from "@zarconontol/enzyme-adapter-react-18";
import Enzyme, { shallow } from "enzyme";
import SetSUbReddit from "./SetSUbReddit";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";

Enzyme.configure({ adapter: new Adapter() });
describe("Component set community info", () => {
  it("should render without crashing", () => {
    expect(
      shallow(
        <SubRedditProvider>
          <SetSUbReddit />
        </SubRedditProvider>
      )
    ).toMatchSnapshot();
  });
});
