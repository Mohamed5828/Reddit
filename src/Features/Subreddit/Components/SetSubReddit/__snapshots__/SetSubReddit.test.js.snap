// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component set community info should render without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "community": null,
      "setCommunity": [Function],
    }
  }
>
  <SetSubReddit />
</ContextProvider>
`;
